### ISSUE TITLE
Implement CMakeDeps configuration to override dependencies information

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan#12367]
From https://github.com/conan-io/conan/issues/12357 from @madebr 

 I think there is a third choice (thanks @jcar87!), that would be much better: if the problem is that we want to allow recipes to patch the behavior of the target names or some other property while generating cmake files, the interface should clearly express that intention, so something like:

```python
def generate(self):
    cmake = CMakeDeps(self)

    cmake.name_alternatives["jansson"] = "Jansson" # overrides the name of the generated files to this
    cmake.target_aliases["jasson::jansson'] = ["Jansson::Jansson"] 
```

This satisfies all needs:
- The changes are clearly and naturally local, they only apply to a local object
- There is no risk of destroying upstream data information

_Originally posted by @memsharded in https://github.com/conan-io/conan/issues/12357#issuecomment-1287188481_

**Comment by prince-chrismc** [comment]
This would also be amazing to inject variables for older projects https://github.com/conan-io/conan-center-index/pull/13551
HDF5 would really benefit form this feature.

**Comment by prince-chrismc** [comment]
I think this can be closed with 1.55? #12609
